packages/aws-cdk-lib/aws-codedeploy/lib/server/deployment-group.ts

```diff
@@ -391,6 +401,39 @@ export class ServerDeploymentGroup extends DeploymentGroupBase implements IServe
     }
   }
 
+  private loadBalancersInfo(loadBalancers?: LoadBalancer[]):
+  CfnDeploymentGroup.LoadBalancerInfoProperty | undefined {
+
+    if (!loadBalancers) {
+      return undefined;
+    }
+
+    // TODO, find a better type here
+    const loadBalancerInfo: {
+      elbInfoList: {name: string}[],
+      targetGroupInfoList: {name: string}[]
+    } = {
+      elbInfoList: [],
+      targetGroupInfoList: [],
+    };
+
+    loadBalancers.forEach(loadBalancer => {
+
+      switch (loadBalancer.generation) {
+        case LoadBalancerGeneration.FIRST:
+          loadBalancerInfo.elbInfoList.push({ name: loadBalancer.name });
+          break;
+        case LoadBalancerGeneration.SECOND:
+          loadBalancerInfo.targetGroupInfoList.push({ name: loadBalancer.name });
+          break;
+        default:
+          throw new Error('Unknown load balancer generation');
+      }
+    });
+
+    return loadBalancerInfo;
+  }
```

Comment:
```suggestion
  private loadBalancersInfo(loadBalancers?: LoadBalancer[]):
  CfnDeploymentGroup.LoadBalancerInfoProperty | undefined {
    if (!loadBalancers) {
      return undefined;
    }

    return loadBalancers.reduce((obj: {
      elbInfoList?: {name: string}[],
      targetGroupInfoList?: {name: string}[]
    }, loadBalancer: LoadBalancer) => {
      switch (loadBalancer.generation) {
        case LoadBalancerGeneration.FIRST:
          if (!obj.elbInfoList) obj.elbInfoList = [];
          obj.elbInfoList.push({ name: loadBalancer.name });
          return obj;
        case LoadBalancerGeneration.SECOND:
          if (!obj.targetGroupInfoList) obj.targetGroupInfoList = [];
          obj.targetGroupInfoList.push({ name: loadBalancer.name });
          return obj;
        default:
          return obj;
      }
    }, {});
  }
```
Cleaner approach using `reduce` and setting the parameters only if a load balancer for the generation is present (may still be improved of course 😄).